File: backend/registrations/deleteRegistration.mjs
import { GetCommand, DeleteCommand } from "@aws-sdk/lib-dynamodb";
import { db, REGISTRATIONS_TABLE } from "./shared/db.mjs";
import { ok, badRequest, notFound, forbidden, serverError, cors } from "./shared/response.mjs";
import { audit, callerFromEvent, isAdmin } from "./shared/auth.mjs";

export async function handler(event) {
  if (event.httpMethod === "OPTIONS") return cors();

  const caller = callerFromEvent(event);
  if (!isAdmin(caller.groups)) return forbidden();

  const { id } = event.pathParameters ?? {};
  if (!id) return badRequest("id is required");

  try {
    const existing = await db.send(new GetCommand({
      TableName: REGISTRATIONS_TABLE,
      Key: { registrationId: id },
    }));
    if (!existing.Item) return notFound("Registration not found");

    await db.send(new DeleteCommand({
      TableName: REGISTRATIONS_TABLE,
      Key: { registrationId: id },
    }));
    await audit("participant.deleted", {
      entity: "participant",
      entityId: id,
      actorId: caller.actorId,
      actorLabel: caller.actorLabel,
      meta: { name: existing.Item.fullName, email: existing.Item.email, phone: existing.Item.phone },
    });
    return ok({ deleted: true });
  } catch (e) {
    console.error(e);
    return serverError();
  }
}

File: backend/registrations/deleteRegistration.py
import logging

from shared.auth import audit, caller_from_event, is_admin
from shared.db import registrations_table
from shared.response import bad_request, cors, forbidden, not_found, ok, server_error, with_cors

log = logging.getLogger()


@with_cors
def handler(event, context):
    if event.get("httpMethod") == "OPTIONS":
        return cors()

    caller = caller_from_event(event)
    if not is_admin(caller["groups"]):
        return forbidden()

    reg_id = (event.get("pathParameters") or {}).get("id")
    if not reg_id:
        return bad_request("id is required")

    try:
        existing = registrations_table().get_item(Key={"registrationId": reg_id}).get("Item")
        if not existing:
            return not_found("Registration not found")

        registrations_table().delete_item(Key={"registrationId": reg_id})
        audit("participant.deleted", entity="participant", entity_id=reg_id,
              actor_id=caller["actorId"], actor_label=caller["actorLabel"],
              meta={"name": existing.get("fullName"), "email": existing.get("email"),
                    "phone": existing.get("phone")})
        return ok({"deleted": True})
    except Exception:
        log.exception("deleteRegistration failed")
        return server_error()
